mantis-control-plane/mantis-control-plane-server/src/main/java/io/mantisrx/master/resourcecluster/ExecutorStateManager.java

```diff
@@ -35,9 +35,23 @@
  * A component to manage the states of {@link TaskExecutorState} for a given {@link ResourceClusterActor}.
  */
 public interface ExecutorStateManager {
-    void putIfAbsent(TaskExecutorID taskExecutorID, TaskExecutorState state);
-    void markAvailable(TaskExecutorID taskExecutorID);
-    void markUnavailable(TaskExecutorID taskExecutorID);
+    /**
+     * Store and track the given task executor's state inside this {@link ExecutorStateManager} if there is no existing
+     * state already. Ignore the given state instance if there is already a state associated with the given ID.
+     * @param taskExecutorID TaskExecutorID
+     * @param state new task executor state
+     * @return whether the given task executor becomes available.
+     */
+    boolean onTaskExecutorStateAssigned(TaskExecutorID taskExecutorID, TaskExecutorState state);
```

Comment:
is the return value ever used?

Reply:
No. It's a refactor on this API return type from void so that it makes more sense to return a bool about whether the operation is successful now after the logic change but the usages currently don't need to use this state.